flake.nix

```diff
@@ -1,24 +1,74 @@
 {
-  description = "Ghaf - documentation for TII SSRC Common Tech";
+  description = "Ghaf - documentation and implementation for TII SSRC Common Tech";
 
-  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
-  inputs.flake-utils.url = "github:numtide/flake-utils";
+  nixConfig = {
+    extra-substituters = [
+      "http://binarycache.vedenemo.dev"
+    ];
+    extra-trusted-public-keys = [
+      "binarycache.vedenemo.dev:Yclq5TKpx2vK7WVugbdP0jpln0/dPHrbUYfsH3UXIps="
+    ];
+  };
 
-  outputs = { self, nixpkgs, flake-utils }:
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-22.11";
+    flake-utils.url = "github:numtide/flake-utils";
+    nixos-generators = {
+      url = "github:nix-community/nixos-generators";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    microvm = {
+      url = "github:astro/microvm.nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.flake-utils.follows = "flake-utils";
+    };
+    jetpack-nixos = {
+      url = "github:anduril/jetpack-nixos";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, nixos-generators, microvm, jetpack-nixos }:
     let
-      systems = [
-        "x86_64-linux"
-        "aarch64-linux"
+      systems = with flake-utils.lib.system; [
+        x86_64-linux
+        aarch64-linux
       ];
     in
-      flake-utils.lib.eachSystem systems (system: {
+    # Combine list of attribute sets together
+    nixpkgs.lib.foldr nixpkgs.lib.recursiveUpdate { } [
+      # Documentation
+      (flake-utils.lib.eachSystem systems (system: {
 
         packages =
           let
             pkgs = nixpkgs.legacyPackages.${system};
-          in {
-            doc = pkgs.callPackage ./doc.nix {};
+          in
+          {
+            doc = pkgs.callPackage ./doc.nix { };
           };
 
-      });
+      }))
+      # Final target images
+      {
```

Comment:
Really nice work with the initial flake. The outputs for the final target images is fine for now but moving forward we could outline host and guest VMs structure as packages (e.g. like doc.nix) so that the top level would only have reference HW packages and those would further build their own hosts but take guest VMs from both common and reference VM specific packages.

Reply:
I put the final target images to a separate file called targets.nix (Is this a good name?). Also added default targets

Reply:
`targets.nix` is a good name.
Please add a commit to include the supported build commands in `ghaf\README.md` like:
https://github.com/mikatammi/ghafhack/blob/master/README.md#x86-64

I tested that:
`nix build` 
now builds the default target fine and results in vm that can be started with `result/bin/run-nixos-vm`

Having this at least in instructions and optionally also in the commit message would be good. I think you can drop `-o result-vm` as the `result` is common with nix.

Reply:
README.md needs some revamp also otherwise, but I can add the section.

Reply:
Also, Ctrl-C does not seem to kill the QEMU process:
```
❯ result/bin/run-nixos-vm
Formatting '/home/vilvo/mikatammi/ghaf/nixos.qcow2', fmt=qcow2 cluster_size=65536 extended_l2=off compression_type=zlib size=1073741824 lazy_refcounts=off refcount_bits=16
^Cqemu: terminating on signal 2
^C
```
Probably worth mentioning in guide if you have a clean method (other than `kill -9 <pid>`) to terminate the VM (e.g. login, `shutdown -h now`). These will be needed by anyone not familiar with the setup and testing it for the first time.

Reply:
> README.md needs some revamp also otherwise, but I can add the section.

It does but we can work that out with @jenninikko

Reply:
> README.md needs some revamp also otherwise, but I can add the section.

You can add just before this https://github.com/tiiuae/ghaf#directory-structure section a new section Build Instructions.

Reply:
I don't think I had those with sway :)

Reply:
There was something else messed with my setup. Now both Ctrl-C work and QEMU window has drop down menu bar. After reboot 🎉

Reply:
I added Build Instructions section into the README.md